### DISCUSSION TITLE
Momentum 4D Object is not subscriptable

### DISCUSSION THREAD

**Discussion by srosenzweig09** [scikit-hep/vector#119] [General]
Hi, I was wondering how to access the p4 value for an entry in a vector.obj object. I have an array of p4 values and I want to access one of them, but if I try to do this, I get an error that the Momentum 4D object is not subscriptable. To get around this, I could loop through my events and build an array of vector.obj objects but that diminishes the power of columnar analysis. I would appreciate any suggestions. Thank you very much!

Example:

```
import uproot
import vector

t = uproot.open("my_file.root:my_tree")
particle_p4 = vector.obj(pt=t['part_pt'], eta=t['part_eta'], phi=t['part_phi'], m=t['part_m'])
...
particle_in_interesting_event = particle_p4[evt_number]
```

This throws an error. Am I trying to do something that is not possible yet? Thanks!

**Comment by jpivarski** [comment]
`vector.obj` makes a single vector object, not an array of vectors. The square brackets/`__getitem__`/subscripting syntax doesn't have a meaning for a single vector.

Maybe what you mean is this:

```python
import awkward as ak
import vector
vector.register_awkward()

pt = t["part_pt"].array()
eta = t["part_eta"].array()
phi = t["part_phi"].array()
m = t["part_m"].array()
particle_p4 = ak.Array({"pt": pt, "eta": eta, "phi": phi, "m": m}, with_name="Momentum4D")
particle_p4  # look at it

all_particles_in_interesting_event = particle_p4[evt_number]
particle_in_interesting_event = all_particles_in_interesting_event[particle_number]
```

or if your data aren't jagged/nested, then only

```python
particle_in_interesting_event = particle_p4[evt_number]
```

would be needed. Such an object would be an Awkward Array with all the usual properties that they have, but the combination of `vector.register_awkward()` and `with_name="Momentum4D"` would also give it Vector methods.

**Reply by jpivarski** [reply]
I know that we _do_ need better documentation on the Vector pages, but maybe this is also pointing to the fact that it also needs error messages when a construction doesn't make sense. If you managed to pass arrays as `pt`, `eta`, `phi` arguments to `vector.obj`, it would be much better if `vector.obj` complained that they're not numbers.

No, wait! `t["part_pt"]` is a TBranch object, not even an array! I'm going to edit my response above so that only arrays are passed to the `ak.Array` constructor.

**Reply by srosenzweig09** [reply]
My apologies for the late response. I should clarify that I load my TBranch objects as awkward tables and numpy tables so they are arrays! Apologies for the confusion.

Yes, I see what you mean, though. I think I often try to use vector.obj objects instead of ak.Array objects and I think this is where I'm going wrong. Thank you!
